src/control/cmd/agent/main.go

```diff
@@ -57,17 +57,69 @@ type cliOptions struct {
 	Insecure   bool       `short:"i" long:"insecure" description:"have agent attempt to connect without certificates"`
 	RuntimeDir string     `short:"s" long:"runtime_dir" description:"Path to agent communications socket"`
 	LogFile    string     `short:"l" long:"logfile" description:"Full path and filename for daos agent log file"`
+	NetworkDbg bool       `short:"n" long:"netdebug" description:"Enable extended network device scan debug output"`
 	Version    versionCmd `command:"version" description:"Print daos_agent version"`
+	NetScan    netScanCmd `command:"scan" description:"Perform local fabric scan"`
 }
 
 type versionCmd struct{}
+type netScanCmd struct {
+	Debug          bool   `short:"d" long:"debug" description:"Enable debug output"`
+	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is all providers)"`
+}
 
 func (cmd *versionCmd) Execute(_ []string) error {
 	fmt.Printf("daos_agent version %s\n", build.DaosVersion)
 	os.Exit(0)
 	return nil
 }
 
+func (cmd *netScanCmd) Execute(args []string) error {
+	var provider string
+	defer os.Exit(0)
+	log := logging.NewCommandLineLogger()
+
+	if cmd.Debug {
```

Comment:
Seconding Mike's comment, this is really confusing. Are -d and -n not already handled by the time we get here?

Reply:
In the current implementation, p.Parse() invokes the Execute handler which never returns.  The opts.* handling for all the options occurs after p.Parse() so they don't have any impact on the sub command that is executed.  So no, -d and -n are not handled by the time we get here.  At least, not the way it is setup currently.

Reply:
Hm, seems like maybe that is a bug in itself, that the subcommand isn't getting the benefit of the top-level flags. I'm in favor of adding a custom command handler like we have in DMG, to make sure the top-level flags are parsed.

Reply:
Thanks Kris.  Now using the top level debug commands.  I've added the CommandHandler override to enable this.